pkg/controller/migration/convert/typha.go

```diff
@@ -0,0 +1,43 @@
+package convert
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/controller/utils"
+)
+
+func checkTypha(c *components, _ *operatorv1.Installation) error {
+	if c.typha != nil {
+		if c.typha.Spec.Replicas == nil {
+			return nil
+		}
+
+		curReplicas := int(*c.typha.Spec.Replicas)
+
+		if curReplicas == 0 {
+			return nil
+		}
+
+		nodes := corev1.NodeList{}
+		err := c.client.List(ctx, &nodes)
```

Comment:
wild idea: should we avoid this client request by checking the status of the existing calico-node daemonset to see how many instances it's running?

Reply:
That is an interesting idea. Though I think there would be issues if operator restarted near when there were only a few of the 'old' calico-node pods left.